src/net_processing.cpp

```diff
@@ -1448,13 +1435,13 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
             return false;
         }
 
-        if (nVersion < nMinPeerProtoVersion)
+        if (nVersion < MIN_PEER_PROTO_VERSION)
         {
             // disconnect from peers older than this proto version
             LogPrintf("peer=%d using obsolete version %i; disconnecting\n", pfrom->id, nVersion);
             connman.PushMessage(pfrom, CNetMsgMaker(INIT_PROTO_VERSION).Make(NetMsgType::REJECT, strCommand, REJECT_OBSOLETE,
-                               strprintf("Version must be %d or greater", nMinPeerProtoVersion)));
-            pfrom->fDisconnect = true;
+                               strprintf("Version must be %d or greater", MIN_PEER_PROTO_VERSION)));
+            Misbehaving(pfrom->GetId(), 100); // don't try to connect again
```

Comment:
Score `100` really a good idea? Someone might just have forgotten to upgrade and will ban himself from all nodes after running this a day or so.

Reply:
Looked into the code before we introduced the auto-disconnect and there was no banning at all. IMHO, we should stick with no banning

Reply:
Oooooops, this shouldn't be here, that's my local thing 🙈